src/services/codefixes/fixReactClassNameAndHTMLFor.ts

```diff
@@ -0,0 +1,60 @@
+/* @internal */
+namespace ts.codefix {
+    const fixID = "fixReactClassNameAndHTMLFor";
+    const errorCodes = [Diagnostics.Type_0_is_not_assignable_to_type_1.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions: context => {
+            const { jsx } = context.program.getCompilerOptions();
+            if (jsx !== JsxEmit.React) {

```

Comment:
Why is this codefix not applicable for `--jsx=preserve`?

Reply:
I didn't detect if developer is using React in their project so it might be an error fix for other JSX frameworks.
But if that JSX framework supports `class` and `for`, this codefix won't appear.
Is the proprably invalid codefix acceptable? 
e.g. I'm using a JSX framework that has no `class` either `className`, this codefix becomes invalid

Reply:
Ah right, I was thinking about this in the context of my other comment about using the type checker to look at the expected props type. I think if you take that approach, it will be safe regardless of JSX emit mode.